services/storehost/storehost.go

```diff
@@ -1558,3 +1558,51 @@ func (t *StoreHost) RemoteReplicateExtent(tCtx thrift.Context, req *store.Remote
 
 	return nil
 }
+
+// ListExtents lists extents available on this storehost
+func (t *StoreHost) ListExtents(tCtx thrift.Context) (res *store.ListExtentsResult_, err error) {
+
+	t.m3Client.IncCounter(metrics.ListExtentsScope, metrics.StorageRequests)
+
+	sw := t.m3Client.StartTimer(metrics.ListExtentsScope, metrics.StorageLatencyTimer)
+	defer sw.Stop()
+
+	log := t.logger
+
+	log.Info("ListExtents()")
+
+	extents, err := t.xMgr.ListExtents()
+
+	if len(extents) > 0 {
+
+		res = store.NewListExtentsResult_()
+
+		res.Extents = make([]*store.ListExtentsElem, len(extents))
+
+		for i, x := range extents {
+
+			info, err := t.xMgr.GetExtentInfo(x)
+
+			if err != nil {
+				continue
+			}
+
+			res.Extents[i] = store.NewListExtentsElem()
+			res.Extents[i].ExtentUUID = common.StringPtr(x)
+			// res.Extents[i].DestinationUUID  = common.StringPtr(...) // TODO: currently unavailable
+			res.Extents[i].Size = common.Int64Ptr(info.Size)
+			res.Extents[i].ModifiedTime = common.Int64Ptr(info.Modified)
+		}
+	}
+
+	if err == nil {
+		return res, nil
+	}
+
+	log.WithField(common.TagErr, err).Error(`ListExtents failed`)
+
+	storeSvcErr := store.NewStoreServiceError()
```

Comment:
see comment earlier about the error handling.. you can move all of this code within if err != nil immediately after the listExtents call.